design/one-pager-package-format-v2.md

```diff
@@ -0,0 +1,302 @@
+  - path: examples/
+```
+
+The above `Configuration` example is exhaustive - it contains all supported
+fields. It should be treated as the authoritative example of Configuration
+metadata. As with the `Provider` example above it is not a Kubernetes custom
+resource. It corresponds to a `Configuration` custom resource in the
+`pkg.crossplane.io` API group. The `meta.pkg.crossplane.io` file is authored by
+the provider maintainer, while the `pkg.crossplane.io` custom resource is
+authored by the platform operator.
+
+### An On-Disk Format
+
+It is likely that there will be multiple on-disk formats; i.e. static files,
+Python scripts, or a [cdk8s] app. This document describes the first such format;
+static files.
+
+This document proposes that:
+
+1. The package contain a `crossplane.yaml` file at its root.
+1. The `crossplane.yaml` file contain either a `Provider` or a `Configuration`.
+1. The package contain zero or more Crossplane specific custom resources (e.g. a
+   `Composition` or a managed resource CRD). Crossplane is not opinionated about
+   where in the package these files exist.
+1. Any extended metadata that is opaque to Crossplane be stored in an
+   appropriately named directory starting with a period - e.g. `.upbound`, or
+   `.registry` - at the root of the package.
+
+Note that 'the package' in this context means both the content of the OCI image
+and the on-disk representation of that content; the OCI image should be built by
+simply copying the content of a directory into a layer of the OCI image. The
+base layer of packages that use this format should specify an entrypoint process
+that will, when invoked:
+
+1. Emit the content of `crossplane.yaml` to stdout.
+1. Discover and emit all Crossplane custom resources contained in the package to
+   stdout, with the exception of those resources ignored by the `ignore` stanza.
+
+The entrypoint may optionally decorate the YAML stream using any extended
+metadata (e.g. the contents of the `.registry` directory). Any such extended
+extended metadata must be written as annotations (`metadata.annotations`) on one
+or more documents in the YAML stream - it may not modify the documents in any
+other way.
+
+### Consuming an Output Stream
+
+Packages must be consumed via their output stream - a YAML stream emitted to
+stdout when the package's OCI image entrypoint is invoked. Package consumers
+must not inspect the content of the OCI image; it should be treated as opaque.
+
+All package entrypoints must support (or silently ignore) the `--context` flag.
+This flag can be used by the package consumer to provide context around how it
+is being consumed; i.e. whether the package is being consumed as part of the
+Crossplane package manager's unpack and install process or by some other system
+such as an indexer. The `--context=unpack` flag will be passed when the package
+manager unpacks a package. The `--context` flag must only affect the annotations
```

Comment:
We can’t actually strictly enforce this, but I like calling this out as convention.